test/js/data/symbol_bucket.test.js

```diff
@@ -72,5 +73,19 @@ test('SymbolBucket', function(t) {
     var b2 = collision.grid.keys.length;
     t.equal(a2, b2, 'detects collision and does not place feature');
 
+    // Use a custom console.warn to count warnings
+    var numWarnings = 0;
+    var warn = console.warn;
+    console.warn = function(warning) {
+        if (warning.includes("Too many symbols being rendered in a tile.") || warning.includes("Too many glyphs being rendered in a tile.")) {
+            numWarnings++;
+        }
+    };
+
+    bucketC.MAX_QUADS = 14;
+    t.equal(bucketC.populateArrays(collision, stacks), undefined);
+    t.equal(numWarnings, 2, 'integer overflow warning is triggered when glyph and/or symbol quad exceeds MAX_QUADS');
+    // Put it back
+    console.warn = warn;
```

Comment:
I copied this approach from [here](https://github.com/mapbox/mapbox-gl-js/blob/master/test/js/data/load_geometry.test.js#L29-L34) after searching for other tests that were looking for warnings. I will look into sinon for a refactor.